### PR TITLE
Allow scala tests on Windows

### PR DESCRIPTION
**Pull request by pimeys**
Before running, read `.envrc` and add the needed environment variables through control panel!

### CODE REVIEW

File: query-engine/connector-test-kit/src/test/scala/util/ConnectorConfig.scala [hunk 1]
```diff
@@ -21,8 +24,15 @@ case class ConnectorConfig(
 object ConnectorConfig {
   lazy val instance: ConnectorConfig = {
     val filePath = EnvVars.serverRoot + "/current_connector"
+
+    val charset = if (EnvVars.isWindows) {
+      StandardCharsets.UTF_16
+    } else {
+      StandardCharsets.UTF_8
+    }
```
Comment (mavilein) [inline query-engine/connector-test-kit/src/test/scala/util/ConnectorConfig.scala:32]: The might not be needed. I would try `Charset.defaultCharset()` and see if this works as well.

File: query-engine/connector-test-kit/src/test/scala/util/TestDatabase.scala [hunk 2]
```diff
@@ -19,7 +19,7 @@ case class MigrationEngine(project: Project) {
 
   def resetAndSetupDatabase(): Unit = {
     import scala.sys.process._
-    val cmd = List(EnvVars.migrationEngineBinaryPath, "cli", "-d", project.dataModelWithDataSourceConfig, "qe-setup")
+    val cmd = List(EnvVars.migrationEngineBinaryPath, "cli", "-d", project.envVar, "qe-setup")
```
Comment (mavilein) [inline query-engine/connector-test-kit/src/test/scala/util/TestDatabase.scala:22]: The name `.envVar` seems confusing in this context. How about we rename it to `dataModelWithDataSourceConfigBase64Encoded`?

File: query-engine/connector-test-kit/src/test/scala/util/TestServer.scala [hunk 3]
```diff
@@ -99,78 +99,46 @@ case class TestServer() extends PlayJsonExtensions with LogSupport {
     val binaryLogLevel   = "RUST_LOG" -> s"query_engine=$logLevel,quaint=$logLevel,query_core=$logLevel,query_connector=$logLevel,sql_query_connector=$logLevel,prisma_models=$logLevel,sql_introspection_connector=$logLevel"
     val log_requests_env = if (log_requests) { "LOG_QUERIES" -> "y" } else { ("", "") }
 
-    val response = (project.isPgBouncer, legacy) match {
```
Comment (mavilein) [inline query-engine/connector-test-kit/src/test/scala/util/TestServer.scala:102]: the setting for pg bouncer is suddenly not needed anymore?

File: query-engine/connector-test-kit/src/test/scala/util/ConnectorConfig.scala (hunk 1 above)
Comment (pimeys) [reply query-engine/connector-test-kit/src/test/scala/util/ConnectorConfig.scala:32]: Uh. Yes.

File: query-engine/connector-test-kit/src/test/scala/util/TestDatabase.scala (hunk 2 above)
Comment (pimeys) [reply query-engine/connector-test-kit/src/test/scala/util/TestDatabase.scala:22]: True.

File: query-engine/connector-test-kit/src/test/scala/util/TestServer.scala (hunk 3 above)
Comment (pimeys) [reply query-engine/connector-test-kit/src/test/scala/util/TestServer.scala:102]: Yeah. It's now part of the connection string. I should take this flag out completely.

**Review by dpetrick — APPROVED** [review]

**Pull request merged**